src/sentry/integrations/manager.py

```diff
@@ -14,6 +15,12 @@
 class IntegrationManager:
     def __init__(self) -> None:
         self.__values: dict[str, type[IntegrationProvider]] = {}
+        self.__domain_integrations: dict[IntegrationDomain, list[str]] = defaultdict(
+            list
+        )  # integrations by domain
+        self.__integration_domains: dict[str, IntegrationDomain] = (
+            {}
+        )  # map of integration to domain
```

Comment:
Minor nit: I think it's fine to generate this on the fly from the list of integration providers instead of storing it on the class. I don't think it'll add _too_ much overhead and reduces our need to remove them when something unregisters.